backend/services/event.py

```diff
@@ -0,0 +1,165 @@
+
+        # Convert entries to a model and return
+        return [entity.to_details_model() for entity in entities]
+
+    def create(self, subject: User, event: EventDetails) -> EventDetails:
+        """
+        Creates a event based on the input object and adds it to the table.
+        If the event's ID is unique to the table, a new entry is added.
+        If the event's ID already exists in the table, raise an exception.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            event: a valid Event model representing the event to be added
+
+        Returns:
+            EventDetails: a valid EventDetails model representing the new Event
+        """
+        self._permission.enforce(subject, "events.create", f"events")
+
+        # Checks if the role already exists in the table
+        if event.id:
+            # Raise exception
+            # should this be changed?
+            event.id = None
+        
+        # Otherwise, create new object
+        event_entity = EventEntity.from_details_model(event)
+
+        # Add new object to table and commit changes
+        self._session.add(event_entity)
+        self._session.commit()
+
+        # Return added object
+        return event_entity.to_details_model()
+
+    def get_from_id(self, id: int) -> EventDetails:
+        """
+        Get the event from an id
+        If none retrieved, a debug description is displayed.
+
+        Parameters:
+            id: a valid int representing a unique event ID
+
+        Returns:
+            Event: Object with corresponding ID
+        """
+
+        # Query the event with matching id
+        entity = self._session.query(EventEntity).get(id)
+
+        # Check if result is null
+        if entity:
+            # Convert entry to a model and return
+            return entity.to_details_model()
+        else:
+            # Raise exception
+            raise EventNotFoundException(id);
+
+    def get_events_from_organization(self, slug: str) -> list[EventDetails]:
+        """
+        Get all the events hosted by an organization with id
+
+        Parameters:
+            slug: a valid str representing a unique Organization slug
+
+        Returns:
+            list[EventDetail]: a list of valid EventDetails models
+        """
+
+        # Query the event with matching organization slug
+        events = self._session.query(EventEntity).filter(EventEntity.organization_slug == slug).all()
+        return [event.to_details_model() for event in events]
+
+    def update(self, subject: User, event: EventDetails) -> EventDetails:
+        """
+        Update the event
+        If none found, a debug description is displayed.
+
+        Parameters:
+            event: a valid Event model
+
+        Returns:
+            EventDetails: a valid EventDetails model representing the updated event object
+        """
+        self._permission.enforce(subject, "events.update", f"events")

```

Comment:
Same comment here, agree this could be based on org.events permission for an entire org.